### PR TITLE
[community] Support passing extra params for executing functions in UCFunctionToolkit

### PR DESCRIPTION
**Pull request by serena-ruan**
Thank you for contributing to LangChain!

- [x] **PR title**: "package: description"
  - Where "package" is whichever of langchain, community, core, experimental, etc. is being modified. Use "docs: ..." for purely docs changes, "templates: ..." for template changes, "infra: ..." for CI changes.
  - Example: "community: add foobar LLM"

Support passing extra params when executing UC functions:
The params should be a dictionary with key EXECUTE_FUNCTION_ARG_NAME, the assumption is that the function itself doesn't use such variable name (starting and ending with double underscores), and if it does we raise Exception.
If invalid params passing to the execute_statement, we raise Exception as well.

- [x] **Add tests and docs**: If you're adding a new integration, please include
  1. a test for the integration, preferably unit tests that do not rely on network access,
  2. an example notebook showing its use. It lives in `docs/docs/integrations` directory.

- [x] **Lint and test**: Run `make format`, `make lint` and `make test` from the root of the package(s) you've modified. See contribution guidelines for more: https://python.langchain.com/docs/contributing/

Additional guidelines:
- Make sure optional dependencies are imported within a function.
- Please do not add dependencies to pyproject.toml files (even optional ones) unless they are required for unit tests.
- Most PRs should not touch more than one package.
- Changes should be backwards compatible.
- If you are adding something to community, do not re-import it in langchain.

If no one reviews your PR within a few days, please @-mention one of baskaryan, efriis, eyurtsev, ccurme, vbarda, hwchase17.

### CODE REVIEW

File: libs/community/langchain_community/tools/databricks/_execution.py [hunk 1]
```diff
@@ -122,16 +130,49 @@ def execute_function(
...
+        and function.input_params.parameters
+        and any(
+            p.name == EXECUTE_FUNCTION_ARG_NAME
+            for p in function.input_params.parameters
+        )
+    ):
+        raise ValueError(
+            "Parameter name conflicts with the reserved argument name for executing "
+            f"functions: {EXECUTE_FUNCTION_ARG_NAME}. "
+            f"Please rename the parameter {EXECUTE_FUNCTION_ARG_NAME}."
```
Comment (harupy) [inline libs/community/langchain_community/tools/databricks/_execution.py:144]: Is renaming the parameter the only solution...?

Comment (serena-ruan) [reply libs/community/langchain_community/tools/databricks/_execution.py:144]: I think it's really rare for such situation, do we need to support other solutions?

Comment (harupy) [reply libs/community/langchain_community/tools/databricks/_execution.py:144]: No, but I just thought renaming a parameter can be challenging sometimes.

Comment (harupy) [reply libs/community/langchain_community/tools/databricks/_execution.py:144]: I see, the value `EXECUTE_FUNCTION_ARG_NAME` is `__executor_args__`, which is unlikely to confict.

**Comment by serena-ruan** [comment]
Hi @baskaryan Do you know why the CI fails to install the dependency? I've added it in the extended_testing_deps.txt file.

**Review by baskaryan — APPROVED** [review]

**Review by harupy — APPROVED** [review]
LGTM!

**Pull request merged**